src/compiler/checker.ts

```diff
@@ -13574,7 +13574,10 @@ namespace ts {
                     return indexTypesIdenticalTo(source, target, kind);
                 }
                 const targetInfo = getIndexInfoOfType(target, kind);
-                if (!targetInfo || targetInfo.type.flags & TypeFlags.Any && !sourceIsPrimitive) {
+                if (!targetInfo ||
+                    targetInfo.type.flags & TypeFlags.Any && !sourceIsPrimitive &&
+                    getSignaturesOfStructuredType(source, SignatureKind.Call).length === 0 &&

```

Comment:
`indexTypesRelatedTo` just takes `Type`s as arguments, so this should just be `getSignaturesOfType`:

```suggestion
                    getSignaturesOfType(source, SignatureKind.Call).length === 0 &&
```